File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hash = {}
        arr =[]
        for i in nums:
            hash[i]=hash.get(i,0)+1
        for n,c in hash.items():
            arr.append([c,n])
        arr.sort(reverse=True)
        ans = [x[1] for x in arr]
        return ans[:k]

File: Data Structures & Algorithms/anagram-groups/submission-3.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
       vector<vector<string>> ans;
       unordered_map<string,vector<string>> hash;
       for(auto& str: strs){
        string k = str;
        sort(k.begin(),k.end());
        hash[k].push_back(str);
       }
       for(auto& [key,value]: hash){
        ans.push_back(value);
       }
       return ans;
    }
};
